src/plugins/visualize/public/application/utils/get_top_nav_config.tsx

```diff
@@ -30,6 +31,14 @@ import { VisualizeConstants } from '../visualize_constants';
 import { getEditBreadcrumbs } from './breadcrumbs';
 import { EmbeddableStateTransfer } from '../../../../embeddable/public';
 
+interface VisualizeCapabilities {
+  createShortUrl?: boolean;
```

Comment:
These feel really odd to me... we're not expecting all of these to be present?  Can they not be defaulted?  We're going to assume they're falsy anyway, no?

Reply:
If an application is consuming its own capabilities, then you can assume they will always be present. If you are consuming another application's capabilities, then you can't assume that they'll be available (the plugin may be disabled, for example)

Reply:
In this case the application is consuming its own capabilities, so I can make the properties required. Thanks both. I'll make the properties required.